includes/cache/SquidUpdate.php

```diff
@@ -122,13 +122,33 @@ static function purge( $urlArr ) {
 			return;
 		}
 
-		global $wgPurgeSquidViaCelery;
+		global $wgPurgeSquidViaCelery, $wgPurgeVignetteUsingSurrogateKeys;
 		if ( $wgPurgeSquidViaCelery == true ) {
-			( new AsyncCeleryTask() )
-					->taskType('celery_workers.purger.purge')
-					->setArgs( $urlArr, [] )
-					->setPriority( PurgeQueue::NAME )
-					->queue();
+			if ( $wgPurgeVignetteUsingSurrogateKeys == true ) {
+				// Filter array for vignette urls and split them into separate task
+				$vignetteUrls = [];
+				foreach ( $urlArr as $index => $url ) {
+					if ( VignetteRequest::isVignetteUrl($url) ) {
+						$vignetteUrls[] = $url;
+						unset($urlArr[$index]);
```

Comment:
It's fine, it does leave "gaps" in the index like 0, 1, 3, 4   But in this case we don't care.